### PR TITLE
feat: finish task 7

### PR DESCRIPTION
**Pull request by antonshchokotov**
### **Task completed.**

**Main scope:**
:white_check_mark: 1 - authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file
:white_check_mark: 3 - import-service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.
:white_check_mark: 5 - update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser localStorage https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage authorization_token = localStorage.getItem('authorization_token')

**Additional scope:**
:white_check_mark: +1 - Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file

-----

**FE: https://d3dr8mns94b5ig.cloudfront.net/**

**importProductsFile lambda** API endpoint: https://ail80uyay7.execute-api.eu-west-1.amazonaws.com/dev/import

-----

**Import file example:** [cycle_store_products_import.zip](https://github.com/antonshchokotov/nodejs-aws-be/files/5579768/cycle_store_products_import.zip)

title,description,price,count,image
Title 1,Description 1,500,1,https://images.unsplash.com/photo-1576435728678-68d0fbf94e91?w=400
Title 2,Description 2,1000,2,https://images.unsplash.com/photo-1456990493443-0d0ee2a630cc?w=400
Title 3,Description 3,1500,3,https://images.unsplash.com/photo-1531727950572-d380a3ae193d?w=400
Title 4,Description 4,2000,4,
Title 5,Description 5,2500,5,test_wrong_url

-----

**credentials** antonshchokotov:TEST_PASSWORD
**token** YW50b25zaGNob2tvdG92OlRFU1RfUEFTU1dPUkQ=

Capture

### CODE REVIEW

**Comment by inshatan** [comment]
Please provide the link to the import-service import lambda

**Comment by antonshchokotov** [comment]
> Please provide the link to the import-service import lambda

Added link, importProductsFile lambda API endpoint: https://ail80uyay7.execute-api.eu-west-1.amazonaws.com/dev/import

File: authorization-service/src/handlers/basicAuthorizer.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const basicAuthorizer = async (event, ctx, cb) => {
+  console.log('function started, event:', event)
+
+  if (event.type !== 'TOKEN') cb('Unauthorized')
+
+  try {
+    const { authorizationToken } = event
+    const encodedCreds = authorizationToken.split(' ')[1]
```
Comment (lexx129) [inline authorization-service/src/handlers/basicAuthorizer.js:8]: It seems that deployed version expects only the token, without anything before it (like 'Basic ' part). I tried 'Basic %encoded_token%, app returned 403 error

Comment (antonshchokotov) [reply authorization-service/src/handlers/basicAuthorizer.js:8]: Lambda authorizer expects authorizationToken in the format "string token" (word|space|token). If Authorization header contains correct token with 'Basic ' part, it works, please double check network tab authorization header.

**Review by Vladlan — APPROVED** [review]

**Pull request merged**